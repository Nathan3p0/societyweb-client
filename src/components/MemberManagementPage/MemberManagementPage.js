import React from 'react';
import MembersList from '../MembersList/MembersList'

const MemberManagementPage = () => {
    return (
        <section className="admin__members-list">
            <MembersList />
        </section>
    );
}

export default MemberManagementPage;