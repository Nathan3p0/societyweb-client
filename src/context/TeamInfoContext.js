import React from 'react';

const TeamInfoContext = React.createContext();

export default TeamInfoContext;