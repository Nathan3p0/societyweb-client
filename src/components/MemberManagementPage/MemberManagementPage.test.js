import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MemberManagementPage from './MemberManagementPage'
import TeamInfoContext from '../../context/TeamInfoContext';

describe('Member Management Page Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<TeamInfoContext.Provider value={{ members: [] }}>
            <MemberManagementPage />
        </TeamInfoContext.Provider>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<TeamInfoContext.Provider value={{ members: [] }}>
            <MemberManagementPage />
        </TeamInfoContext.Provider>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})