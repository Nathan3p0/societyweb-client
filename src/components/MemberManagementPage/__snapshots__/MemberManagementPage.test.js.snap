// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Management Page Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="admin__members-list"
>
  <section
    className="membersList__main"
  >
    <h3>
      Current Members:
    </h3>
    <ul
      className="responsiveTable"
    >
      <li
        className="tableHeader"
      >
        <div
          className="col col-1"
        >
          Full Name
        </div>
        <div
          className="col col-2"
        >
          Username
        </div>
        <div
          className="col col-3"
        >
          Phone Number
        </div>
        <div
          className="col col-4"
        >
          Email
        </div>
      </li>
      <p
        className="error"
      />
    </ul>
  </section>
</section>
`;
